src/Criterion.cpp

```diff
@@ -24,6 +24,19 @@ Criterion::Criterion(const Criterion &crit){
     weight_ = crit.getWeight();
 }
 
+void Criterion::generateDirection(unsigned long int seed){ 
+    srand(seed);
+    if ( ((float) rand() / RAND_MAX) < 0.5){ direction_ = -1; }
+    else { direction_ = 1; }
+}
+
+void Criterion::generateWeight(unsigned long int seed){
+    srand(seed);
+    weight_ = (float) rand() / RAND_MAX ;
+}
+
+
+
```

Comment:
Should we have one extra function that combines both in order to generate a random Criterion with one method function call ?

Reply:
We could implement such functions indeed, maybe let's wait to analyze the different algorithms to check which initializations are we going to use?

Reply:
That function would be useful for the dataGenerator im working on for example 👍

Reply:
and where we could control the randomness with a seed in order to make test work

Reply:
will be solved in Thibault's generator